sql/sql_acl.cc

```diff
@@ -14181,29 +14186,29 @@ static bool acl_check_ssl(THD *thd, const ACL_USER *acl_user)
   */
   switch (acl_user->ssl_type) {
   case SSL_TYPE_NOT_SPECIFIED:                  // Impossible
-  case SSL_TYPE_NONE:                           // SSL is not required
+  case SSL_TYPE_NONE:                           // SSL is not required FOR THIS SPECIFIC USER
     if (opt_require_secure_transport)
     {
       enum enum_vio_type type= vio_type(vio);
-#ifdef HAVE_OPENSSL
-      return type != VIO_TYPE_SSL &&
-#ifndef _WIN32
-             type != VIO_TYPE_SOCKET;
-#else
-             type != VIO_TYPE_NAMEDPIPE;
-#endif
-#else
-#ifndef _WIN32
-      return type != VIO_TYPE_SOCKET;
-#else
-      return type != VIO_TYPE_NAMEDPIPE;
-#endif
-#endif
+
+      bool has_secured_transport =
+      #ifdef HAVE_OPENSSL
```

Comment:
note the VIO_TYPE defines aren't limited by defines at all. Also Wolfssl is a SSL implementation. This could be a simple 3 condition `||` test.  You can use `#ifdef _WIN32` around the named pipe.

Reply:
I looked closer, use of `#ifdef` seems common around these macros and `HAVE_OPENSSL` is defined for Wolfssl (that provides a compatible interface).

so fine as is.